client/service/src/builder.rs

```diff
@@ -205,12 +208,27 @@ pub type TLightClientWithBackend<TBl, TRtApi, TExecDisp, TBackend> = Client<
 	TRtApi,
 >;
 
-enum KeystoreContainerInner {
-	Local(Arc<LocalKeystore>)
+trait AsCryptoRef {
+    fn keystore_ref(&self) -> Arc<dyn CryptoStore>;
+    fn sync_keystore_ref(&self) -> Arc<dyn SyncCryptoStore>;
+}
+
+struct DoubleWrap<T>(Arc<T>);
+
+impl<T> AsCryptoRef for DoubleWrap<T> where T: CryptoStore + SyncCryptoStore + 'static {
```

Comment:
Question: What happens if we get rid of `DoubleWrap` and implement `AsCryptoRef` on `Arc<T>` directly? the below methods could return `self.clone()` in that case and the code would be a bit simpler.

I have modified this snippet you showed me earlier to do this:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5d6a9deb6c38c00cab9d0198b95d690c